### DISCUSSION TITLE
FYI; "Matterbridge zigbee2mqtt" is a Matter bridge plug-in/add-on for the Zigbee2MQTT open-souce Zigbee gateway

### DISCUSSION THREAD

**Discussion by Hedda** [matter-js/matter.js#803] [General]
For your information; "**_Matterbridge zigbee2mqtt_**" is a new open-source plug-in/add-on project by @Luligu for the very popular open-source [Zigbee2MQTT](https://www.zigbee2mqtt.io/) (DIY Zigbee gateway) allowing bridging of native Zigbee devices to a Matter fabric,  exposing them as Matter devices:

* https://github.com/Luligu/matterbridge-zigbee2mqtt

Since I understand that the Matter standard specification is actually based on the same [ZCL (Zigbee Cluster Library) specification](https://zigbeealliance.org/wp-content/uploads/2021/10/07-5123-08-Zigbee-Cluster-Library.pdf) as Zigbee 3.0 from Connectivity Standards Alliance (CSA), so native Matter devices are basically using the same or similar clusters and attributes, so translating Zigbee devices into Matter should in theory be relatively easy. The problem today though is that the released versions of the Matter standard do not yet expose as many different device types as the ZCL (Zigbee Cluster Library) specification does for Zigbee 3.0 devices.

PS: There is related discussion in this feature request for Zigbee2Matter (bridging Zigbee to Matter) as a Z2M Matter bridge subproject:

https://github.com/Koenkk/zigbee2mqtt/discussions/7443

**Comment by Hedda** [comment]
For end-user reference; a Matter bridge is a device that connects another smart home ecosystem to a user's Matter fabric (your Matter network at home). Thus you will need to have a compatible Matter controller that is compatible matter.js/matter-node.js, see:

* https://github.com/project-chip/matter.js?tab=readme-ov-file#compatibility

Anyway, these types of third-party Matter bridges will serve a very important role in Matter smart home and its widespread adoption, as they will allow users to bring in their existing collection of non-Matter smart home gear into other Matter compatible ecosystems. 

This allows users to connect their Zigbee devices directly to a Matter fabric through a middleman. The non-Matter devices are exposed as Bridged Devices to Nodes in the Matter ecosystem. The Bridge device performs the translation between other protocols and Matter so that the Matter nodes can communicate with the Bridged Devices.

### Matter Bridge:

![image](https://user-images.githubusercontent.com/6320001/229044369-b3c5b59d-3166-4621-9794-49eddc65e748.png)

Especially low-power devices that natively support the "Matter" standard over Thread are still rare (and expensive) so I guess most people who already have a large setup with Zigbee (or  Z-Wave) devices in their existing environment will not want to immediately start over to buy new native Matter devices or switch to using devices that use another mesh networking standard and thus those people who still want to use Zigbee  (and Zwave) devices but want to utilize Matter will want home automation gateways/hubs that acts as "Zigbee to Matter" bridges (and "Z-Wave to Matter" bridges).

### Matter-Zigbee Bridge Workflow:

![image](https://user-images.githubusercontent.com/6320001/229044603-265f424f-91bb-42df-96df-a3ea0ed1cf6a.png)

Zigbee to Matter Bridge example 

###  Below is an example of Matter-Zigbee Bridge that bridges two Zigbee lights to Matter ecosystem:

![image](https://github.com/Koenkk/zigbee2mqtt/assets/6320001/93c2dafb-06c9-4125-ba98-e2dcf16b647d)

PS: As a commercial example, check out the existing Philips Hue Hub appliance recieved a firmware update late last year that made it Matter compatible as a Matter bridge, and that also makes it a Zigbee to Matter bridge that links Philips Hue Hub user's existing Zigbee-based Philips Hue lights and accessories into their Matter ecosystem. While none of Philips's existing lights and accessories products will  their own ever become Matter compatible natively, but through the Matter bridge feature, they will happily play along with the rest of a user's other Matter gear:

https://www.philips-hue.com/en-us/explore-hue/works-with/matter

**Comment by Luligu** [comment]
Hi,
I'm the author of https://github.com/Luligu/matterbridge and all the example plugins.
The https://github.com/Luligu/matterbridge-zigbee2mqtt plugin (production level) does the best possible but:
```
Out of the box, this plugin supports all possible conversion from zigbee2mqtt to Matter 1.1.

Since the Matter support in the available ecosystems (controllers) is very limited and, when available, only covers Matter 1.1 specifications, some z2m devices cannot be exposed properly or cannot be exposed at all.
```
On HA we also expose the electrical measurements (Voltage, Current, Consumption and TotalConsumption).

**Reply by Hedda** [reply]
> On HA we also expose the electrical measurements (Voltage, Current, Consumption and TotalConsumption).

I believe upcoming Matter 1.3 Release Milestone should bring energy consumption monitoring as standard among other things?

* https://github.com/project-chip/connectedhomeip/milestone/65
* https://github.com/project-chip/connectedhomeip/milestone/54

- Electrical Power Measurement
- Electrical Energy Measurement
- Water Heater Management

* https://github.com/project-chip/connectedhomeip/issues/30193
* https://github.com/project-chip/connectedhomeip/issues/30264
  * https://github.com/project-chip/connectedhomeip/issues/31037
* https://github.com/project-chip/connectedhomeip/issues/30268

**Reply by Apollon77** [reply]
Right, but then it will need practically 6+ months until this reaches devices and controllers ... usually
